started coops functions

jsta opened the pull request:
I thought I would start on #111. I hope the code style is ok. I tried to mirror the work in 8202b8aeb4241e0f588fc22526b248c7842f1ae5.

sckott commented:
sweet :+1:  , I'll have a look

sckott commented:
@jsta Can you install latest roxygen2 https://cran.rstudio.com/web/packages/roxygen2/ and re-run `document` or similar

sckott commented on R/coops.R:
```diff
@@ -0,0 +1,28 @@
+#' @param begin_date numeric date in yyyymmdd format
+#' @param end_date numeric date in yyyymmdd format
+#' @param datum character choice of MHHW, MHW, MTL, MSL, MLW, MLLW, NAVD, STND
+#' @param product character choice of water_level, air_temperature, water_temperature, wind, air_pressure, air_gap, conductivity, visibility, humidity, salinity, hourly_height, high_low, daily_mean, monthly_mean, one_minute_water_level, predictions, datums, currents.
```
@jsta please shorten lines to less than 100 so easier to read, and check doesn't complain

sckott commented on R/coops.R:
```diff
@@ -0,0 +1,28 @@
+#' @examples \dontrun{
+#' coops_search(station_name = 8723970, begin_date = 19820301, end_date = 20141001, datum = "stnd", product = "monthly_mean")
+#' }
+coops_search <- function(begin_date = NULL, end_date = NULL, station_name = NULL, product = NULL, datum = NULL, units = "metric", time_zone = "gmt", application = "web_services", format = "json"){
```
@jsta it appears that 4 params don't have param definitions `units`, `time_zone`, `application`, `format`, can you add those?

sckott replied:
@jsta please add a `...` parameter in the function definition, and a param def for it, so users can pass in curl options, see e.g., https://github.com/ropensci/rnoaa/blob/master/R/ghcnd.R#L7

sckott commented on R/coops.R:
```diff
@@ -0,0 +1,28 @@
+
+  args <- noaa_compact(list(begin_date = begin_date, end_date = end_date, station = station_name, product = product, datum = datum, units = units, time_zone = time_zone, application = application, format = format))
+
+  coops_GET(coops_base(), args)
```
@jsta change to `coops_GET(coops_base(), args, ...)`

sckott commented on R/coops.R:
```diff
@@ -0,0 +1,28 @@
+#' @param product character choice of water_level, air_temperature, water_temperature, wind, air_pressure, air_gap, conductivity, visibility, humidity, salinity, hourly_height, high_low, daily_mean, monthly_mean, one_minute_water_level, predictions, datums, currents.
+#' @export
+#' @references \url{http://co-ops.nos.noaa.gov/api/}
+#' @examples \dontrun{
```
@jsta can you add more examples?  so users (and I) can get a better idea of how to use the various params

sckott commented:
@jsta Please do add a test file for this function, including some tests to make sure the function fails well

jsta commented:
Ok, I think I correctly fixed the documentation, added examples, and added tests. I am not too confident in writing tests so I hope I did ok. @sckott 

Other things to work on include:

* what to put for the default application parameter in the args to `coops_GET()`
* enable users to simply request either a start or end date and a range

sckott commented:
@jsta thanks

> what to put for the default application parameter in the args to coops_GET()

you could leave it blank if you want to force users to give a value for it, instead of you choosing it, thoughts?

> enable users to simply request either a start or end date and a range 

do you want to do that before I merge, or do you plan to do it later?

jsta commented:
Do you not agree that `application` should be set at the `rnoaa` level  and not be user set? The argument against this might be that a users who integrate `rnoaa` as a dependency would want to set their own app as `application`. The [api instructions](http://co-ops.nos.noaa.gov/api/) say:

> Application
> If you are a CO-OPS developer, please include the application name calling the API. If you are an external user, please provide the name of your organization for this parameter when calling this API. 

I think adding a range option can be done later.

jsta commented:
I make a persuasive argument to have it user-set :)

sckott commented:
> Do you not agree that application should be set at the rnoaa level and not be user set? The argument against this might be that users who integrate rnoaa as a dependency would want to set their own package as application. 

I think what  you did makes sense, to have a default of the name of the pkg. 

> I think adding a range option can be done later.

great, sounds good.
